src/include/duckdb/parser/column_definition.hpp

```diff
@@ -12,31 +12,74 @@
 #include "duckdb/common/types/value.hpp"
 #include "duckdb/parser/parsed_expression.hpp"
 #include "duckdb/common/enums/compression_type.hpp"
+#include "duckdb/catalog/catalog_entry/table_column_info.hpp"
+#include "duckdb/common/case_insensitive_map.hpp"
 
 namespace duckdb {
 
+struct RenameColumnInfo;
+struct RenameTableInfo;
+
+enum ColumnExpressionType { DEFAULT, GENERATED };
+
+struct ColumnExpression {
+	unique_ptr<ParsedExpression> expression;
+	ColumnExpressionType type;
+	ColumnExpression(unique_ptr<ParsedExpression> expression, ColumnExpressionType type = ColumnExpressionType::DEFAULT)
+	    : expression(move(expression)), type(type) {
+	}
+};
+
+class ColumnDefinition;
+
+//! Used in both TableCatalogEntry and TransformCreateTable
+void AddToColumnDependencyMapping(ColumnDefinition &col, case_insensitive_map_t<unordered_set<string>> &dependents,
+                                  case_insensitive_map_t<unordered_set<string>> &dependencies);
+
 //! A column of a table.
 class ColumnDefinition {
 public:
 	DUCKDB_API ColumnDefinition(string name, LogicalType type);
-	DUCKDB_API ColumnDefinition(string name, LogicalType type, unique_ptr<ParsedExpression> default_value);
+	DUCKDB_API ColumnDefinition(string name, LogicalType type, ColumnExpression expression);
```

Comment:
`TableColumnType` and expression -> if generated, move expression to generated_expression, otherwise move to default_value?